### PR TITLE
Several fixes for Rpm patcher and Combiner

### PR DESCRIPTION
**Pull request by a-lyupa**

### CODE REVIEW

**Review by rayslava — COMMENTED** [review]
Is `/etc/hosts` enough?
AFAIR `mic` copies a set of files from `/etc` like `resolv.conf` and so on.

File: combirepo/rpm_patcher.py [hunk 1]
```diff
@@ -668,15 +664,41 @@ def __process_results(self):
...
+            temporaries.umount_image(self.patching_root)
+            if "modules.img" in self.images_dict:
+                temporaries.umount_image(os.path.join(self.patching_root,
+                                                      self.images_dict["modules.img"]))
+
+    def __mount_fs(self):
+        """
+        Mount system directories required for patching.
+        """
+        self.mount_points = ["sys", "proc", "dev", "dev/pts", "dev/null"]
```
Comment (rayslava) [inline combirepo/rpm_patcher.py]: I'd add `/dev/mqueue` and `/dev/shm` as well.

File: combirepo/rpm_patcher.py [hunk 2]
```diff
@@ -732,12 +754,14 @@ def do_tasks(self):
...
                 self.__use_cached_root_or_prepare()
                 self.__clone_chroots()
+                self.__umount_root()
+                self.__mount_fs()
                 self.__deploy_packages()
                 hidden_subprocess.function_call_monitor(
                     self.__patch_packages, (), self._status_callback)
                 self.__postprocess_cache()
                 self.__process_results()
-                self.__umount_root()
```
Comment (rayslava) [inline combirepo/rpm_patcher.py:738]: And is root left mounted?

File: jenkins/combirepo_run.jenkinsfile [hunk 3]
```diff
@@ -316,7 +316,7 @@ pipeline {
           -o ${micDir} -l ${logFile} --tmp-dir ${cacheDir} -j ${params.JOBS_NUMBER} \
           \${greedy} \${mirror} \${rpm_patching} \${debug} \${verbose} \
           --regenerate-repodata --preferring-strategy big --skip-version-mismatch --mic-config ${micFile} \
-          -S libasan,asan-runtime-env,make,cpio,rpm-build --disable-libasan-preloading \
+          -S libasan,asan-runtime-env --mic-options="--pack-to=snapshot.tar" \
```
Comment (rayslava) [inline jenkins/combirepo_run.jenkinsfile]: `libasan` is required by `asan-runtime-env`, I think the latter package should be enough.

File: combirepo/rpm_patcher.py (hunk 2 above)
Comment (a-lyupa) [reply combirepo/rpm_patcher.py:738]: No, I moved this line a little above.

File: jenkins/combirepo_run.jenkinsfile (hunk 3 above)
Comment (a-lyupa) [reply jenkins/combirepo_run.jenkinsfile]: Ok, I'll check it.

File: combirepo/rpm_patcher.py (hunk 1 above)
Comment (a-lyupa) [reply combirepo/rpm_patcher.py]: Ok, I'll add.

Comment (a-lyupa) [reply combirepo/rpm_patcher.py]: > Is /etc/hosts enough?
> AFAIR mic copies a set of files from /etc like resolv.conf and so on.

I'll check it. I'm not sure if it's enough.

Comment (a-lyupa) [reply combirepo/rpm_patcher.py]: Mic writes to /etc/hosts, /etc/resolv.conf and some of /etc/sysconfig/ files. I'd pass hostname through kickstart file.

Comment (a-lyupa) [reply combirepo/rpm_patcher.py]: I removed copying hosts. I'll try to pass it through kickstart file later. It's not critical.

Comment (a-lyupa) [reply combirepo/rpm_patcher.py]: Added more mount points.

File: jenkins/combirepo_run.jenkinsfile (hunk 3 above)
Comment (a-lyupa) [reply jenkins/combirepo_run.jenkinsfile]: Only asan-runtime-env is OK.

**Review by rayslava — COMMENTED** [review]
Please add some tag (e.g. [Submodule] to rpmrebuild commit.

**Review by rayslava — APPROVED** [review]

**Pull request merged**